src/components/Button/Button.vue

```diff
@@ -0,0 +1,152 @@
+<template>
+  <a
+    v-if="href"
+    v-bind="buttonProps"
+    :href="href"
+    v-on="$listeners"
+  >
+    <div>
+      <span
+        v-if="title"
+        v-bind="titleProps"
+      >{{ title }}</span>
+      <slot v-else />
+    </div>
+  </a>
+  <nuxt-link
+    v-else-if="to"
+    v-bind="buttonProps"
+    :to="to"
+    v-on="$listeners"
+  >
+    <div>
+      <span
+        v-if="title"
+        v-bind="titleProps"
+      >{{ title }}</span>
+      <slot v-else />
+    </div>
+  </nuxt-link>
+  <button
+    v-else
+    v-bind="buttonProps"
+    v-on="$listeners"
+  >
+    <div>
+      <span
+        v-if="title"
+        v-bind="titleProps"
+      >{{ title }}</span>
+      <slot v-else />
+    </div>
+  </button>
+</template>
+
+<script>
+export default {
+  name: 'Button',
+  props: {
+    preset: {
+      type: String,
+      default: 'primary',
+    },
+    href: {
+      type: String,
+      default: '',
+    },
+    title: {
+      type: String,
+      default: '',
+    },
+    to: {
+      type: Object,
+      default: undefined,
+    },
+  },
+  computed: {
+    buttonProps() {
+      return {
+        class: ['button', `button--${this.preset}`],
+        ...this.$attrs,
+      };
+    },
+    titleProps() {
+      return {
+        class: 'button__title',
+      };
+    },
+  },
+};
+</script>
+
+<style lang="scss">
+.button {
+  display: inline-block;
+
+  padding: 0;
+
+  min-height: 32px;
+  min-width: 32px;
+
+  border: none;
+  border-radius: 8px;
+
+  text-align: center;
+  text-decoration: none;
+
+  > div {
+    display: inline-flex;
+    box-sizing: border-box;
+    justify-content: center;
+    align-items: center;
+
+    min-height: inherit;
+    min-width: inherit;
+    padding: 4px;
+
+    border-radius: inherit;
+
+    transition: background-color 0.2s ease;
+  }
+
+  &:hover > div {
+    background-color: #00000008;
+  }
+
+  &:active > div {
+    background-color: #0000001a;
+  }
+
+  &__title {
+    padding: 0 8px;
+
+    font-family: Source Sans Pro, Arial, sans-serif;
```

Comment:
Should `Source Sans Pro, Arial, sans-serif` be in some kind of variable, instead of appearing in different components?

Reply:
I think we are going to do it in a theme provider component in the future, I think this will do for now